br/pkg/task/stream.go

```diff
@@ -1252,38 +1252,27 @@ func getGlobalResolvedTS(
 	ctx context.Context,
 	s storage.ExternalStorage,
 ) (uint64, error) {
-	storeMap := make(map[int64]uint64)
-
-	opt := &storage.WalkOption{SubDir: restore.GetStreamBackupMetaPrefix()}
-	err := s.WalkDir(ctx, opt, func(path string, size int64) error {
-		if strings.Contains(path, restore.GetStreamBackupMetaPrefix()) {
-			m := &backuppb.Metadata{}
-			b, err := s.ReadFile(ctx, path)
-			if err != nil {
-				return errors.Trace(err)
-			}
-			err = m.Unmarshal(b)
-			if err != nil {
-				return errors.Trace(err)
-			}
-
-			if resolveTS, exist := storeMap[m.StoreId]; !exist || resolveTS < m.ResolvedTs {
-				storeMap[m.StoreId] = m.ResolvedTs
-			}
+	storeMap := struct {
+		sync.Mutex
+		resolvedTSMap map[int64]uint64
+	}{}
+	storeMap.resolvedTSMap = make(map[int64]uint64)
+	err := stream.FastUnmarshalMetaData(ctx, s, func(path string, m *backuppb.Metadata) error {
+		if resolveTS, exist := storeMap.resolvedTSMap[m.StoreId]; !exist || resolveTS < m.ResolvedTs {
+			storeMap.resolvedTSMap[m.StoreId] = m.ResolvedTs
```

Comment:
Need mutex here to protect resolvedTSMap

Reply:
Yes, I'll fix it.